### PR TITLE
Adding Ssh Client Support

### PR DESCRIPTION
**Pull request by deep1712**

### CODE REVIEW

File: src/VirtualClient/VirtualClient.Common/SSHClient/ISshClientProxy.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace VirtualClient.Common
+{
+    using System;
+    using Renci.SshNet;
```
Comment (yangpanMS) [inline src/VirtualClient/VirtualClient.Common/SSHClient/ISshClientProxy.cs:7]: Do you know how popular this package is? Is there something from .NET?

File: src/VirtualClient/VirtualClient.Common/SSHClient/ISshClientProxy.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        /// Remote connection information of the SshClient.
+        /// </summary>
+        ConnectionInfo ConnectionInfo { get; }
+
+        /// <summary>
+        /// Creates the command to be executed.
+        /// </summary>
+        /// <param name="commandText">The command text.</param>
+        /// <returns>A Ssh Command object.</returns>
+        ISshCommandProxy CreateCommand(string commandText);
```
Comment (yangpanMS) [inline src/VirtualClient/VirtualClient.Common/SSHClient/ISshClientProxy.cs:24]: Is it create command or run command?

**Comment by yangpanMS** [comment]
Which workloads or monitors need this?

File: src/VirtualClient/VirtualClient.Common/SSHClient/ISshClientProxy.cs (hunk 2 above)
Comment (deep1712) [reply src/VirtualClient/VirtualClient.Common/SSHClient/ISshClientProxy.cs:24]: It just creates a command. We can run the commands independently.

File: src/VirtualClient/VirtualClient.Common/SSHClient/ISshClientProxy.cs (hunk 1 above)
Comment (deep1712) [reply src/VirtualClient/VirtualClient.Common/SSHClient/ISshClientProxy.cs:7]: image

Comment (deep1712) [reply src/VirtualClient/VirtualClient.Common/SSHClient/ISshClientProxy.cs:7]: 77.8 million downloads in total for library.

**Comment by yangpanMS** [comment]
@deep1712 could you have a discussion with me about what this could be used for?

File: src/VirtualClient/Module.props [hunk 3]
```diff
@@ -128,6 +128,9 @@
         <!-- Serilog.Sinks.RollingFile -->
         <Serilog_Sinks_RollingFile_PackageVersion>3.3.0</Serilog_Sinks_RollingFile_PackageVersion>
 
+		<!-- SSH.Net -->
+		<SSH_Net_PackageVersion>2020.0.2</SSH_Net_PackageVersion>
```
Comment (yangpanMS) [inline src/VirtualClient/Module.props:132]: nit: spacing

File: src/VirtualClient/VirtualClient.Actions/HPCG/HpcgExecutor.cs [hunk 4]
```diff
@@ -135,11 +136,11 @@ private async Task CaptureMetricsAsync(IProcessProxy process, EventContext telem
                 {
                     foreach (string file in outputFiles)
                     {
-                        string contents = await this.LoadResultsAsync(file, cancellationToken);
+                        string results = await this.LoadResultsAsync(file, cancellationToken);
```
Comment (yangpanMS) [inline src/VirtualClient/VirtualClient.Actions/HPCG/HpcgExecutor.cs:139]: those renames are fine, just making sure these are intended

File: src/VirtualClient/VirtualClient.Common/IProcessProxy.cs [hunk 5]
```diff
@@ -37,10 +37,15 @@ public interface IProcessProxy : IDisposable
         int ExitCode { get; }
 
         /// <summary>
-        /// 
+        /// Gets the exit time for the underlying process.
```
Comment (yangpanMS) [inline src/VirtualClient/VirtualClient.Common/IProcessProxy.cs:40]: Thank you!

**Review by yangpanMS — APPROVED** [review]

**Pull request merged**